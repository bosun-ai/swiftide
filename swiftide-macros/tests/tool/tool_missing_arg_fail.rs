#[swiftide_macros::tool(
    description = "My first tool",
    param(name = "msg", description = "A message for testing")
)]
async fn basic_tool(
    _agent_context: &dyn AgentContext,
    msg: &str,
    other: &str,
) -> Result<ToolOutput, ToolError> {
    Ok(format!("Hello {msg}").into())
}

#[swiftide_macros::tool(
    description = READ_FILE,
    param(name = "number", description = "Number to guess")
)]
async fn guess_a_number(
    _context: &dyn AgentContext,
    number: usize,
) -> Result<ToolOutput, ToolError> {
    let actual_number = 42;

    if number == actual_number {
        Ok("You guessed it!".into())
    } else {
        Ok("Try again!".into())
    }
}
fn main() {}
