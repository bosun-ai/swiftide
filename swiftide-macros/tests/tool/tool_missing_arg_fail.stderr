error: Arguments in spec and in function do not match:
        The following parameters are missing from the spec: ["other"]
 --> tests/tool/tool_missing_arg_fail.rs:1:1
  |
1 | / #[swiftide_macros::tool(
2 | |     description = "My first tool",
3 | |     param(name = "msg", description = "A message for testing")
4 | | )]
  | |__^
  |
  = note: this error originates in the attribute macro `swiftide_macros::tool` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Params that are not strings need to have their `type` as json spec specified
  --> tests/tool/tool_missing_arg_fail.rs:13:1
   |
13 | / #[swiftide_macros::tool(
14 | |     description = READ_FILE,
15 | |     param(name = "number", description = "Number to guess")
16 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `swiftide_macros::tool` (in Nightly builds, run with -Z macro-backtrace for more info)
